src/mbgl/renderer/paint_parameters.cpp

```diff
@@ -72,8 +72,10 @@ gfx::DepthMode PaintParameters::depthModeForSublayer(uint8_t n, gfx::DepthMaskTy
     return gfx::DepthMode { gfx::DepthFunctionType::LessEqual, mask, { nearDepth, farDepth } };
 }
 
-gfx::DepthMode PaintParameters::depthModeFor3D(gfx::DepthMaskType mask) const {
-    return gfx::DepthMode { gfx::DepthFunctionType::LessEqual, mask, { 0.0, 1.0 } };
+gfx::DepthMode PaintParameters::depthModeFor3D() const {
+    return gfx::DepthMode{ gfx::DepthFunctionType::LessEqual,
+                           gfx::DepthMaskType::ReadWrite,
+                           { 0.0, depthRangeSize } };
```

Comment:
Question, `depthRangeSize` is [set after](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/renderer_impl.cpp#L428) 3D pass, near / far range would be `{0.0f, 0.0f}`, right?

Reply:
Yeah, this is a misnomer. The "3D" pass doesn't actually render anything related to 3D anymore. It's only used for heatmap and hillshading preprocessing, but not for fill extrusions anymore (we should rename it). The `depthModeFor3D` function isn't called in 3D pass. I'll keep the name as is to match JS' naming.

Reply:
:+1: thanks for explanation